napari/components/add_layers_mixin.py

```diff
@@ -737,7 +795,21 @@ def _add_layer_from_data(
 
         """
 
-        layer_type = layer_type.lower()
+        layer_type = (layer_type or '').lower()
+
+        # this came from __main__.py
+        # assumes that big integer type arrays are likely labels.
+        if not layer_type:
+            if hasattr(data, 'dtype') and data.dtype in (
+                np.int32,
+                np.uint32,
+                np.int64,
+                np.uint64,
+            ):
+                layer_type = 'labels'
+            else:
+                layer_type = 'image'
```

Comment:
I know this is preserving previous behaviour, that I think @jni added, but I don't really like this special treatment of labels - i think fine to assume if `layer_type` not provided it is an image, but i don't like guessing that it is labels. We can add a `layer_type` flag to our CLI if that is needed.

Reply:
that might be tricky though as it means we need to add one to `add_path` and its unclear how to pass it in a nice way, only if it hasn't been provided by the reading

Reply:
Well, on the one hand, the plugins allow `layer_type` to be declared now (this was not the case pre-plugins), so it's fair to make that the canonical mechanism. On the other, though, I don't recall literally *ever* seeing a normal (ie intensity, not label) image stored as one of those dtypes. Ever. Have you? So to me it seems like, if a plugin wants to do the weird thing and load such an image as intensities, it can return the `'image'` layer type, and otherwise, we do our inference...

Reply:
Yeah, I'm not opposed to letting an un-specified uint32/64 layer be coerced to labels unless they specifically provide `image` layer_type.  I've never seen those dtypes used for anything else.

Reply:
fine with me